### ISSUE TITLE
support self joins in the SQL feature provider

### ISSUE THREAD

**Issue by cportele** [interactive-instruments/ldproxy#503]

**Comment by azahnen** [comment]
@cportele This should work as expected in `xtraplatform-spatial` version `5.0.0-sql-queries-SNAPSHOT`.

**Comment by cportele** [comment]
@azahnen - I have tested it with an updated gebaeude configuration (I have also updated the repo) and get an `IndexOutOfBoundsException` in `SqlMultiplicityTracker.java:70`. Could you check, if you get that, too?

**Comment by azahnen** [comment]
@cportele Yes, I get that as well. I tested with a version where `consistsOfBuildingPart` just contained links, which works fine. Using the same joins again a level deeper does not work yet in `SqlMultiplicityTracker` since it does not use the full paths as identifier. I will fix that later today.

**Comment by azahnen** [comment]
@cportele I pushed a fix. It took me a while to find out that further issues occurred due to duplicate primary roles in the mapping. I pushed a fix for that as well.

**Comment by cportele** [comment]
Great, thanks.

**Comment by cportele** [comment]
@azahnen - In my tests I have found three other issues that I have fixed through commits in the `sql-queries` branch, but please check, if my changes are correct. I have also noticed that not all cases have been covered in the `getFilter()` function.

Commits:
* https://github.com/interactive-instruments/xtraplatform-spatial/commit/caa6f16face9effd0eaf944a7edb3d3585095991
* https://github.com/interactive-instruments/xtraplatform-spatial/commit/c8cea0c6cb5e67dfb6149d5b2d871bdcbb85a9f9
* https://github.com/interactive-instruments/xtraplatform-spatial/commit/2de621a5c3f4dd7acc31bd4e98212000b7c3831b
